src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1014,24 +1014,99 @@ public static void Format_Invalid()
+        [InlineData("____________________________\ufffd", '\ufffd', 0, 29, 28)]
+        [InlineData("ABCDEFGHIJKLM", 'M', 0, 13, 12)]
+        [InlineData("ABCDEFGHIJKLMN", 'N', 0, 14, 13)]
+        [InlineData("ABCDEFGHIJKLMNOPQRSTUVWXYZ", '@', 0, 26, -1)]
+        [InlineData("ABCDEFGHIJKLMNOPQRSTUVWXY", '@', 0, 25, -1)]
+        [InlineData("ABCDEFGHIJKLMNOPQRSTUVWXYZ#", '@', 0, 27, -1)]
+        [InlineData("_____________\u807f", '\u007f', 0, 14, -1)]
+        [InlineData("_____________\u807f__", '\u007f', 0, 16, -1)]
+        [InlineData("_____________\u807f\u007f_", '\u007f', 0, 16, 14)]
+        [InlineData("__\u807f_______________", '\u007f', 0, 18, -1)]
+        [InlineData("__\u807f___\u007f___________", '\u007f', 0, 18, 6)]
+        [InlineData("ABCDEFGHIJKLMN", 'N', 2, 11, -1)]
+        [InlineData("!@#$%^&", '%', 0, 7, 4)]
+        [InlineData("!@#$", '!', 0, 4, 0)]
+        [InlineData("!@#$", '@', 0, 4, 1)]
+        [InlineData("!@#$", '#', 0, 4, 2)]
+        [InlineData("!@#$", '$', 0, 4, 3)]
+        [InlineData("!@#$%^&*", '%', 0, 8, 4)]
         public static void IndexOf_SingleLetter(string s, char target, int startIndex, int count, int expected)
         {
+            bool safeForCurrentCulture =
+                IsSafeForCurrentCultureComparisons(s)
+                && IsSafeForCurrentCultureComparisons(target.ToString());
+
             if (count + startIndex == s.Length)
             {
                 if (startIndex == 0)
                 {
                     Assert.Equal(expected, s.IndexOf(target));
-                    Assert.Equal(expected, s.IndexOf(target.ToString()));
+                    Assert.Equal(expected, s.IndexOf(target.ToString(), StringComparison.Ordinal));
+                    Assert.Equal(expected, s.IndexOf(target.ToString(), StringComparison.OrdinalIgnoreCase));
+
+                    // To be safe we only want to run CurrentCulture comparisons if
+                    // we know the results will not vary depending on location
+                    if (safeForCurrentCulture)
+                    {
+                        Assert.Equal(expected, s.IndexOf(target.ToString()));
+                        Assert.Equal(expected, s.IndexOf(target.ToString(), StringComparison.CurrentCulture));
+                    }
                 }
                 Assert.Equal(expected, s.IndexOf(target, startIndex));
-                Assert.Equal(expected, s.IndexOf(target.ToString(), startIndex));
+                Assert.Equal(expected, s.IndexOf(target.ToString(), startIndex, StringComparison.Ordinal));
+                Assert.Equal(expected, s.IndexOf(target.ToString(), startIndex, StringComparison.OrdinalIgnoreCase));
+
+                if (safeForCurrentCulture)
+                {
+                    Assert.Equal(expected, s.IndexOf(target.ToString(), startIndex));
+                    Assert.Equal(expected, s.IndexOf(target.ToString(), startIndex, StringComparison.CurrentCulture));
```

Comment:
@stephentoub I just tested by putting

```cs
CultureInfo.CurrentCulture = new CultureInfo("tr-TR");
```

at the beginning of the method. The answer seems to be yes. (These tests only run if all the chars in the string are ASCII and not control.)